cltk/text_reuse/automata.py

```diff
@@ -0,0 +1,452 @@
+"""
+Constructs Levenshtein automata
+
+At first, both the NFA and DFA classes inherited from a common Automaton class.
+I found that treating the two as separate entities, results in more comprehensive
+documentation in the price of succinctness.
+
+This is the third iteration of the algorithm, the first one used transition
+matrices for representing automata but this proved to be a massive waste
+of space for anything non-trivial. In addition, instead of the usual
+intuitiveness that accompanies such approaches, I had to deal with
+countless of indexing errors and even the slightest of optimizations
+became overbearing (e.g. minimizing an automaton and deleting unreachable
+states, required either costly operations or error-prone hacks).
+
+After a brief experimenting with more traditional graph approaches,
+I eventually tried trie-like nested dictionaries. This came with
+a whole set of new problems (can't use unhashable types as dictionary,
+traversing a nested dictionary is not as direct as a matrix etc), but
+the overall code is more understandable for someone already familiar
+with the subject matter.
+"""
+
+__author__ = ['Eleftheria Chatziargyriou <ele.hatzy@gmail.com>']
+__license__ = 'MIT License. See LICENSE.'
+
+
```

Comment:
Since you do some print outputs, prefer to use our standard API logging approach add the logging import and initialization code:
 
import logging

LOG = logging.getLogger(__name__)
LOG.addHandler(logging.NullHandler())